src/frontend/src/handler/explain.rs

```diff
@@ -34,35 +35,23 @@ use crate::optimizer::OptimizerContext;
 use crate::scheduler::BatchPlanFragmenter;
 use crate::stream_fragmenter::build_graph;
 use crate::utils::explain_stream_graph;
+use crate::OptimizerContextRef;
 
-pub async fn handle_explain(
-    handler_args: HandlerArgs,
+async fn do_handle_explain(
+    context: OptimizerContext,
     stmt: Statement,
-    options: ExplainOptions,
-    analyze: bool,
-) -> Result<RwPgResponse> {
-    let context = OptimizerContext::new(handler_args.clone(), options.clone());
-
-    if analyze {
-        return Err(ErrorCode::NotImplemented("explain analyze".to_string(), 4856.into()).into());
-    }
-
-    let session = context.session_ctx().clone();
-
-    let mut plan_fragmenter = None;
-    let mut rows = {
-        let plan = match stmt {
-            Statement::CreateView {
-                or_replace: false,
-                materialized: true,
-                query,
-                name,
-                columns,
-                ..
-            } => gen_create_mv_plan(&session, context.into(), *query, name, columns)?.0,
+    blocks: &mut Vec<String>,
+) -> Result<()> {
+    // Workaround to avoid `Rc` across `await` point.
+    let mut batch_plan_fragmenter = None;
```

Comment:
Actually, most of the time I spent on this PR is struggling with `Rc` and `Send`. 🥵 I guess either we need a refactor to split blocking steps off the planning, or we may migrate to thread-safe structures to make developing easier.

Reply:
:+1: :heart:  :hot_face:  https://github.com/risingwavelabs/risingwave/issues/9515